File: 0860-lemonade-change/0860-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int fives=0,tens=0,len=bills.length;
        for(int i=0;i<len;i++){
            if(bills[i]==5) fives++;
            else if(bills[i]==10){
                if(fives==0) return false;
                fives--;
                tens++;
            }
            else if(tens>0 && fives>0){
                tens--;
                fives--;
            }
            else if(fives>=3) fives-=3;
            else return false;
        }
        return true;
    }
}

File: 0860-lemonade-change/0860-lemonade-change.kt
class Solution {
fun lemonadeChange(bills: IntArray): Boolean {
    var five = 0
    var ten = 0
    for (bill in bills) {
        when (bill) {
            5 -> five++
            10 -> {
                if (five == 0) return false
                five--
                ten++
            }
            else -> {
                if (five > 0 && ten > 0) {
                    five--
                    ten--
                } else if (five >= 3) {
                    five -= 3
                } else {
                    return false
                }
            }
        }
    }
    return true
}
}